### PR TITLE
add documentation for reading lower-dimensional objects

### PR DESCRIPTION
**Pull request by keflavich**
Solves #833 

I smuggled in a configuration item - `ext=0` is now a keyword arg instead of hard-coding it.

### CODE REVIEW

File: docs/creating_reading.rst [hunk 1]
```diff
@@ -35,6 +35,60 @@ accessing other Stokes parameters, see :doc:`stokes`.
...
+------------------------
+
+While spectral-cube is designed for cube analysis, in the course of normal
+analysis procedures you are likely to need to load up one- and two-dimensional
+subsets or views of the data.
+
+You can load :class:`~spectral_cube.lower_dimensional_objects.Projection`
+objects from FITS HDU objects with
+:meth:`~spectral_cube.lower_dimensional_objects.Projection.from_hdu`.  Only
+FITS reading is supported::
```
Comment (e-koch) [inline docs/creating_reading.rst:49]:
```suggestion
FITS reading is currently supported::
```
Raised a separate issue for loading 2D CASA images: https://github.com/radio-astro-tools/casa-formats-io/issues/47

File: docs/creating_reading.rst [hunk 2]
```diff
@@ -35,6 +35,60 @@ accessing other Stokes parameters, see :doc:`stokes`.
...
+
+    >>> from astropy.io import fits  # doctest: +SKIP
+    >>> hdul = fits.open('file.fits')
+    >>> projection = Projection.from_hdu(hdul)
+
+Note that if you pass in a :class:`astropy.io.fits.HDUList` object, data will be loaded
```
Comment (e-koch) [inline docs/creating_reading.rst:55]:
```suggestion
Note that if you pass in a :class:`astropy.io.fits.HDUList` object, by default the data will be loaded
```

File: docs/creating_reading.rst [hunk 3]
```diff
@@ -35,6 +35,60 @@ accessing other Stokes parameters, see :doc:`stokes`.
...
+from the first HDU; this can be overridden with the ``ext`` keyword.
```
Comment (e-koch) [inline docs/creating_reading.rst:56]:
```suggestion
from the first HDU. To load a different HDU in the list, the index can be passed to the ``ext`` keyword (e.g., `ext=1` to load the second HDU in the list).
```

File: spectral_cube/lower_dimensional_structures.py [hunk 4]
```diff
@@ -367,13 +367,13 @@ def _new_projection_with(self, data=None, wcs=None, mask=None, meta=None,
         return newproj
 
     @staticmethod
-    def from_hdu(hdu):
+    def from_hdu(hdu, ext=0):
         '''
         Return a projection from a FITS HDU.
         '''
```
Comment (e-koch) [inline spectral_cube/lower_dimensional_structures.py:373]:
```suggestion
    Parameters
    -----------
    ext : int
        The integer index to load when given an :class:`astropy.io.fits.HDUList`.
        Default is 0 (the first HDU in the list.

        '''
```

File: spectral_cube/lower_dimensional_structures.py [hunk 5]
```diff
@@ -625,14 +625,14 @@ def __repr__(self):
         return '{0}{1}{2:s}>'.format(prefixstr, arrstr, self._unitstr)
 
     @staticmethod
-    def from_hdu(hdu):
+    def from_hdu(hdu, ext=0):
         '''
         Return a OneDSpectrum from a FITS HDU or HDU list.
         '''
```
Comment (e-koch) [inline spectral_cube/lower_dimensional_structures.py:631]:
```suggestion
    Parameters
    -----------
    ext : int
        The integer index to load when given an :class:`astropy.io.fits.HDUList`.
        Default is 0 (the first HDU in the list.

        '''
```

**Review by e-koch — COMMENTED** [review]
Looks good! Just a few text suggestions.

File: spectral_cube/lower_dimensional_structures.py (hunk 5 above)
Comment (keflavich) [reply spectral_cube/lower_dimensional_structures.py:631]: indentation error; will commit & fix

File: spectral_cube/lower_dimensional_structures.py [hunk 6]
```diff
@@ -625,14 +625,20 @@ def __repr__(self):
...
-    def from_hdu(hdu):
+    def from_hdu(hdu, ext=0):
         '''
         Return a OneDSpectrum from a FITS HDU or HDU list.
+    Parameters
+    -----------
+    ext : int
+        The integer index to load when given an :class:`astropy.io.fits.HDUList`.
+        Default is 0 (the first HDU in the list.
+
```
Comment (keflavich) [inline spectral_cube/lower_dimensional_structures.py:636]:
```suggestion

        Parameters
        -----------
        ext : int
            The integer index to load when given an :class:`astropy.io.fits.HDUList`.
            Default is 0 (the first HDU in the list.

```

File: spectral_cube/lower_dimensional_structures.py [hunk 7]
```diff
@@ -367,13 +367,19 @@ def _new_projection_with(self, data=None, wcs=None, mask=None, meta=None,
...
+    def from_hdu(hdu, ext=0):
         '''
         Return a projection from a FITS HDU.
+    Parameters
+    -----------
+    ext : int
+        The integer index to load when given an :class:`astropy.io.fits.HDUList`.
+        Default is 0 (the first HDU in the list.
+
         '''
```
Comment (keflavich) [inline spectral_cube/lower_dimensional_structures.py:379]:
```suggestion
        Return a projection from a FITS HDU.

        Parameters
        -----------
        ext : int
            The integer index to load when given an :class:`astropy.io.fits.HDUList`.
            Default is 0 (the first HDU in the list.

        '''
```

**Comment by keflavich** [comment]
This probably ought to be squashed on merge because of all the gh-suggest commits

**Review by e-koch — APPROVED** [review]

**Comment by keflavich** [comment]
afaict, the error is unrelated.  I'm gonna merge and hope the windows-based, dask-test error just goes away...

**Pull request merged**